crates/interfaces/src/executor.rs

```diff
@@ -1,12 +1,14 @@
 use async_trait::async_trait;
-use reth_primitives::{Block, Bloom, H256};
+use reth_primitives::{Address, Block, Bloom, H256};
 use thiserror::Error;
 
-/// Takes block and executes it, returns error
+/// An executor capable of executing a block.
 #[async_trait]
-pub trait BlockExecutor: Send + Sync {
-    /// Execute block
-    async fn execute(&self, _block: Block) -> Error;
+pub trait BlockExecutor<T> {
+    /// Execute a block.
+    ///
+    /// The number of `senders` should be equal to the number of transactions in the block.
```

Comment:
```suggestion
    /// The number of `senders` should be equal to the number of transactions in the block.
    ///
    /// If no senders are specified, the `execute` function MUST recover the senders for the
    /// provided block's transactions internally. We use this to allow for calculating senders in
    /// parallel in e.g. staged sync, so that execution can happen without paying for sender recovery costs.
```